openstudiocore/src/bimserver/ProjectImportation.cpp

```diff
@@ -0,0 +1,347 @@
+/**********************************************************************
+*  Copyright (c) 2008-2015, Alliance for Sustainable Energy.
+*  All rights reserved.
+*
+*  This library is free software; you can redistribute it and/or
+*  modify it under the terms of the GNU Lesser General Public
+*  License as published by the Free Software Foundation; either
+*  version 2.1 of the License, or (at your option) any later version.
+*
+*  This library is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+*  Lesser General Public License for more details.
+*
+*  You should have received a copy of the GNU Lesser General Public
+*  License along with this library; if not, write to the Free Software
+*  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+**********************************************************************/
+
+#include "ProjectImportation.hpp"
+#include "BIMserverConnection.hpp"
+
+#include "../osversion/VersionTranslator.hpp"
+
+
+#include <QFileDialog>
+#include <QGridLayout>
+#include <QVBoxLayout>
+#include <QMessageBox>
+#include <QLabel>
+#include <QEventLoop>
+
+#include <QLineEdit>
+#include <QPushButton>
+#include <QDebug>
+
+namespace openstudio {
+namespace bimserver {
+
+	ProjectImportation::ProjectImportation(QWidget *parent) :
```

Comment:
@zhouchong90 Importation is 'the act of importing', if possible I would prefer to rename this class 'ProjectImporter' instead